### ISSUE TITLE
Type error in getTopSameDomainWindow()

### ISSUE THREAD

**Issue by bancer** [DevExpress/testcafe#7886]
### What is your Scenario?

After upgrading from TestCafe 2.6.2 to 3.0.1 some tests failed with this stacktrace:

![Screenshot from 2023-07-18 16-08-07](https://github.com/DevExpress/testcafe/assets/3830106/cfd99e1f-78af-4852-bdca-88d800bf9600)

Repeating the test steps manually in the browser produced no errors.

The error was gone after adding `disableNativeAutomation: true,` to the configuration file.

### What is the Current behavior?

See above.

### What is the Expected behavior?

I would expect no error.

### What is your public website URL? (or attach your complete example)

I can't produce an example as I see no relationship between our test and the error stacktrace.

### What is your TestCafe test code?

```
test('Add new guest', async t => {
    'use strict';
    await t
        .maximizeWindow()
        .useRole(admin)
        .navigateTo(url)
    ;
    await selectOptionFromDropdownMenu('#InitialsOverlaySelect', 'CT');
    await t
        .click('#guestDatabase img')
        .click('#action-add-customer')
        .typeText(new Selector('#CustomerFirstName').nth(1), 'James')
        .typeText(new Selector('#CustomerLastName').nth(1), 'McGill')
        .typeText(new Selector('#CustomerEmail').nth(1), 'jamesmcgillTest@example.com')
        .typeText(new Selector('#CustomerAddress').nth(1), 'Alex')
        .click('#addRestaurantCustomerButton')
        .typeText('#GuestFirstName', 'James')
        .typeText('#GuestLastName', 'McGill')
        .doubleClick(new Selector('#GuestDatabaseTable td').withText('James'))
        .expect(new Selector('#RestaurantCustomerFirstName').nth(1).value)
        .eql('James')
        .expect(new Selector('#RestaurantCustomerLastName').nth(1).value)
        .eql('McGill')
    ;
});
```

### Your complete configuration file

```
var TestCafeModule = {
    browsers: "chrome:headless",
    cache: true,
    // Quarantine mode hides unstable tests and increases the overal tests running time
    quarantineMode: false,
    stopOnFirstFail: true,
    src: "tests/UserInterfaceTests/desktop",
    disableNativeAutomation: true,
    screenshots: {
        takeOnFails: true,
        path: "tmp/img/",
        pathPattern: "${DATE}-${TIME}-${TEST_INDEX}-${FILE_INDEX}-${QUARANTINE_ATTEMPT}.png",
        thumbnails: false,
    },
};
// support setting multiple reporters like "dashboard,spec-time" which we use on CI UI tests
let rep = process.env.TESTCAFE_REPORTER ? process.env.TESTCAFE_REPORTER : "spec-time";
const myRep = rep.split(",");
let reporters = [];
for (var i = 0; i < myRep.length; i++) {
    reporters[i] = {
        name: myRep[i],
    };
}
TestCafeModule.reporter = reporters;
module.exports = TestCafeModule;
```

### Your complete test report

Output in the console was:
``` 
✖ Add new guest (8.21s)

   1) A JavaScript error occurred on "http://...".
      Repeat test actions in the browser and check the console for errors.
      Enable the “skipJsErrors” option to ignore JavaScript errors during test execution. Learn more: "https://testcafe.io/documentation/404038/recipes/debugging/skip-javascript-errors"
      If the website only throws this error when you test it with TestCafe, please create a new issue at:
      "https://github.com/DevExpress/testcafe/issues/new?template=bug-report.md".

      JavaScript error details:
      Script error.
          No stack trace available

      Browser: Chrome 114.0.0.0 / Ubuntu 20.04

         13 |    await selectOptionFromDropdownMenu('#InitialsOverlaySelect', 'CT');
         14 |    await t
         15 |        .click('#guestDatabase img')
         16 |        .click('#action-add-customer')
         17 |        .typeText(new Selector('#CustomerFirstName').nth(1), 'James')
       > 18 |        .typeText(new Selector('#CustomerLastName').nth(1), 'McGill')
         19 |        .typeText(new Selector('#CustomerEmail').nth(1), 'jamesmcgillTest@example.com')
         20 |        .typeText(new Selector('#CustomerAddress').nth(1), 'Alex')
         21 |        .click('#addRestaurantCustomerButton')
         22 |        .typeText('#GuestFirstName', 'James')
         23 |        .typeText('#GuestLastName', 'McGill')

         at <anonymous> (/.../tests/UserInterfaceTests/desktop/guest.js:18:10)
         at asyncGeneratorStep (/.../tests/UserInterfaceTests/desktop/guest.js:2:86)
         at _next (/.../tests/UserInterfaceTests/desktop/guest.js:2:86)

```

### Steps to Reproduce

1.
2.
3.

### TestCafe version

3.0.1

### Node.js version

v18.16.1

### Command-line arguments

testcafe -- --test-grep 'Add new guest'

### Browser name(s) and version(s)

Chrome 114.0.0.0

### Platform(s) and version(s)

Ubuntu 20.04 LTS

**Comment by bancer** [comment]
A clue could be that the HTML is not valid having duplicate ids hence `new Selector('#CustomerFirstName').nth(1)` in the test selecting the 2nd element.

**Comment by Aleksey28** [comment]
Hi @bancer,

The described behavior looks like a bug in TestCafe. We need to reproduce and research it on our side. Please share with us the public URL of your web site with some test credentials so that we can log in and find the cause of the issue.

Note that our policy prevents us from accessing internal resources without prior written approval from the site owner. If you want us to research the problem directly on your side, please ask the website owner to send us ([support@devexpress.com](mailto:support@devexpress.com)) written confirmation. It must permit the DevExpress personnel to remotely access the website and its internal resources for research, testing, and debugging purposes.

**Comment by bancer** [comment]
@Aleksey28 I can't give you access to the page but I opened a pull request with a fix - https://github.com/DevExpress/testcafe-hammerhead/pull/2936.

**Comment by miherlosev** [comment]
See https://github.com/DevExpress/testcafe-hammerhead/pull/2936#issuecomment-1647115073

**Comment by ayemelyanenko-chegg** [comment]
Sorry to chime in but does the testcafe log help in reproducing?

https://upload.disroot.org/r/1x3h34w8#bdMOCO33y7ltfgPrMsgsN9EyQcDBqSfc2pBzA6FApbk=

**Comment by andrzej-woof** [comment]
It does happen also on the examples page https://react-dropzone.js.org/#section-basic-example (no special steps, just navigate to that url in testcafe with native automation enabled)

**Comment by jgoldsmith613** [comment]
Just a heads up, we are seeing this exact error too.  But we were using native automation in 2.6 already and it passed. But once upgrading to 3.x we are seeing the exact same error on a few cases.  Disabling native automation does fix it, but it must be something that changed with native automation between 2.6 and 3.x

**Comment by Aleksey28** [comment]
Hi @jgoldsmith613,

We already have a [PR](https://github.com/DevExpress/testcafe/pull/7934) with the fix. It will be published in the next release after the merge.

**Comment by ayemelyanenko-chegg** [comment]
@miherlosev Hello, is this fix going to be in the v3.2.0-rc.1?

https://github.com/DevExpress/testcafe/pull/7944

**Comment by miherlosev** [comment]
Yes.
